src/main/scala/com/databricks/spark/avro/AvroRelation.scala

```diff
@@ -50,6 +53,13 @@ private[avro] class AvroRelation(
     private val parameters: Map[String, String])
     (@transient val sqlContext: SQLContext) extends HadoopFsRelation with Logging {
 
+  // This UUID is used to avoid output file name collision between different appending write jobs.
+  // These jobs may belong to different SparkContext instances. Concrete data source implementations
+  // may use this UUID to generate unique file names (e.g., `part-r-<task-id>-<job-uuid>.parquet`).
+  //  The reason why this ID is used to identify a job rather than a single task output file is
+  // that, speculative tasks must generate the same output file name as the original task.
+  private val uniqueWriteJobId = UUID.randomUUID().toString
```

Comment:
Hm, that's weird. The UUID is set [here] [1]. Could you please provide a snippet that reproduces the issue you mentioned? I'll also try to reproduce it locally.

On the other hand, adding the UUID here isn't appropriate, because we'd like to use the UUID to identify each write job rather than each relation. We don't want any two write job have the same UUID.

[1]: https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/WriterContainer.scala#L88-L91